app/controllers/HomeController.scala

```diff
@@ -54,11 +58,20 @@ class HomeController @Inject() (
   private val authenticate: ActionBuilder[UserRequest, AnyContent] =
     authJourney.authWithPersonalDetails
 
+  private def getTaxComponentsOrEmptyList(nino: Nino, year: Int)(implicit
```

Comment:
Using Connectors directly in controller might not be good idea specially, where it is doing some data parsing here. But we can do it later if time urgency is there.

Reply:
I know what you mean. I removed the service layer as it only had one method in it + wasn't doing anything other than passing straight over to connector so it seemed unnecessary. The data parsing is actually done in the connector (passed in from controller as a reads)

Reply:
No need for an empty service.
If you reinstate the service you will need to keep the connector doing only connector thingy.
So move the reads and possibly toggle in the service.
It a simple connector so I am not worried about it. It's fine as it is.

Reply:
So just checking: you want me to create service, move toggle into it from connector, and reads from controller?